core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
@davies I think this is best left undocumented. But is this setting meaningful? This has more purpose than the previous `spark.rdd.enableParallelListing` config, which only let you disable it. I actually don't see a case where you wouldn't want parallel listing (see previous discussion)?  why would i want to list 9 serially but 10 all at once?